extensions/markdown-language-features/src/features/documentLinkProvider.ts

```diff
@@ -102,18 +104,57 @@ export function stripAngleBrackets(link: string) {
 	return link.replace(angleBracketLinkRe, '$1');
 }
 
-const linkPattern = /(\[((!\[[^\]]*?\]\(\s*)([^\s\(\)]+?)\s*\)\]|(?:\\\]|[^\]])*\])\(\s*)(([^\s\(\)]|\([^\s\(\)]*?\))+)\s*(".*?")?\)/g;
+const codeSpanAndLinkPattern = /(?:(?<!`)(`+)(?!`)(?:.+?|.*?(?:(?:\r?\n).+?)*?)(?:\r?\n)?(?<!`)\1(?!`))|(?:(\[((!\[[^\]]*?\]\(\s*)([^\s\(\)]+?)\s*\)\]|(?:\\\]|[^\]])*\])\(\s*)(([^\s\(\)]|\([^\s\(\)]*?\))+)\s*(".*?")?\))/g;
 const referenceLinkPattern = /(\[((?:\\\]|[^\]])+)\]\[\s*?)([^\s\]]*?)\]/g;
 const definitionPattern = /^([\t ]*\[(?!\^)((?:\\\]|[^\]])+)\]:\s*)([^<]\S*|<[^>]+>)/gm;
 
+/**
+ * Asserts if number is inside at least one interval in the array of intervals.
+ *
+ * @param intervals An array of [a,b) sorted by `a` in ascending order.
+ */
+const isNumberInIntervals = (intervals: [number, number][], start: number, end: number, target: number): Boolean => {
```

Comment:
I know binary searches are theoretically better, but honestly I'd just use `some`:

```js
intervals.some(pair => target >= pair[0] && target < pair[1]);
```

Even if you have a document with 1000 code blocks, that will be plenty fast